### PR TITLE
Use _attr_native_value instead of _attr_state in sensor entity

### PR DESCRIPTION
**Pull request by natekspencer**
The `state` property of a sensor entity is marked as final in Home Assistant code and should not be used for a sensor's state. Instead the recommendation is to use the `native_value` property. This switches the code to use `_attr_native_value` instead of `_attr_state` as well as some additional cleanup in the sensor.py file for readability.

See https://github.com/home-assistant/core/blob/dev/homeassistant/components/sensor/__init__.py#L538
and https://developers.home-assistant.io/docs/core/entity/sensor

### CODE REVIEW

File: custom_components/tapo_control/sensor.py [hunk 1]
```diff
@@ -1,104 +1,87 @@
...
-            for hdd in entry["camData"]["sdCardData"]:
-                for sensorProperty in hdd:
-                    LOGGER.debug(
-                        f"Adding TapoHDDSensor for disk {hdd['disk_name']} and property {sensorProperty}..."
-                    )
-                    sensors.append(
-                        TapoHDDSensor(
-                            entry, hass, entry, hdd["disk_name"], sensorProperty
+        if "camData" in entry:
+            cam_data: dict = entry["camData"]
```
Comment (JurajNyiri) [inline custom_components/tapo_control/sensor.py:32]: Please rename to camData to be consistent with other entities

File: custom_components/tapo_control/sensor.py [hunk 2]
```diff
@@ -1,104 +1,87 @@
...
-                    )
 
         sensors.append(TapoSyncSensor(entry, hass, config_entry))
 
         return sensors
 
-    sensors = await setupEntities(entry)
-    for childDevice in entry["childDevices"]:
-        sensors.extend(await setupEntities(childDevice))
+    sensors = await setup_entities(entry)
```
Comment (JurajNyiri) [inline custom_components/tapo_control/sensor.py:66]: Please rename back to setupEntities to be consistent

File: custom_components/tapo_control/sensor.py [hunk 3]
```diff
@@ -1,104 +1,87 @@
...
+    for child_device in entry["childDevices"]:
```
Comment (JurajNyiri) [inline custom_components/tapo_control/sensor.py:67]: Please rename back to be consistent

File: custom_components/tapo_control/sensor.py [hunk 4]
```diff
@@ -109,32 +92,31 @@ def __init__(self, entry: dict, hass: HomeAssistant, config_entry):
             None,
         )
 
-    @property
-    def entity_category(self):
-        return EntityCategory.DIAGNOSTIC
-
     async def async_update(self) -> None:
+        """Update the entity."""
```
Comment (JurajNyiri) [inline custom_components/tapo_control/sensor.py:96]: Remove comment

**Review by JurajNyiri — CHANGES_REQUESTED** [review]
Thank you for the PR! Left some minor comments, once addressed I will test and merge

File: custom_components/tapo_control/sensor.py (hunk 1 above)
Comment (natekspencer) [reply custom_components/tapo_control/sensor.py:32]: Will do! I was trying to follow python naming standards (and resolve pylint complaining in my IDE) and planned on making this change to all other entities, but wanted to start off with just one. But if you'd rather keep it this way instead I'll revert!

**Comment by JurajNyiri** [comment]
Thank you for the reminder and PR, I was very busy lately. I will try to review and test this during the weekend.

**Comment by natekspencer** [comment]
> Thank you for the reminder and PR, I was very busy lately. I will try to review and test this during the weekend.

No worries, just didn't want it to get lost!

**Comment by JurajNyiri** [comment]
Thank you! Merged

**Pull request merged**